### PR TITLE
Add API endpoints with authentication

### PR DESCRIPTION
**Pull request by MReyna12**

### CODE REVIEW

**Comment by MReyna12** [comment]
For the reviewer:

1. Instead of using `first()` as part of the database query, I opted for `all()`. It seemed like an easy way to test the exception `MultipleResultsFound`. Alternatively, we have a guardrail up via the email unique constraint, so maybe we don't need to check for the database query returning more than one user. If that's the case, we could roll back to using `first()` and use the conditional that confirms we actually got a result from our query.

2. I setup two database related fixtures. One clears the database each time the tests are run. The other fixture brings in the database so that we can add our test user to the database. I am unsure if there is a better way to bring in the database to our test file outside of how I did it. The examples in the `FastAPI` docs discuss creating a test database, but that's done using `mysql`. I didn't see a "simple" `postgreSQL` alternative. Feel free to suggest any alternative for implementing the database in the test file.

Lastly, I did add `pytest-cov` to the repo and the coverage report did return back 100% coverage for the `test_endpoints.py` file.

**Comment by rnathuji** [comment]
@MReyna12 - I've noted this in some initial review comments, but just in case I don't get a chance to finish that today due to meetings sharing here too. You'll want to run the following to get a coverage report on the Python code:

```bash
$ pytest --cov=rope --cov-report=term --cov-report=html
============================================================================== test session starts ===============================================================================
platform linux -- Python 3.10.13, pytest-8.0.0, pluggy-1.4.0
rootdir: /home/rnathuji/openstax/src/rope/backend
plugins: mock-3.12.0, anyio-4.2.0, cov-4.1.0
collected 4 items                                                                                                                                                                

tests/test_endpoints.py ....                                                                                                                                               [100%]

---------- coverage: platform linux, python 3.10.13-final-0 ----------
Name              Stmts   Miss  Cover
-------------------------------------
api/__init__.py       0      0   100%
api/auth.py          21      4    81%
api/database.py      19      2    89%
api/main.py          39      2    95%
api/sessions.py      10      1    90%
api/settings.py       3      0   100%
db/__init__.py        0      0   100%
db/schema.py         15      0   100%
-------------------------------------
TOTAL               107      9    92%
Coverage HTML written to dir htmlcov
```

:point_up: should give you a summary in the terminal, and you can open up details with line-by-line views in `./htmlcov/index.html`.

**Comment by MReyna12** [comment]
Ahhh I was looking at the pytest-cov docs and I think only tested the endpoint file. That would explain the discrepancy. Thanks!

File: backend/tests/test_endpoints.py [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+        "rope.api.sessions.sessions",
+        user,
+    )
+    response = test_client.get("/user/current")
+    assert response.status_code == 401
+
+
+def test_delete_session(test_client, mocker):
+    app.dependency_overrides[get_request_session] = override_get_request_session
+    mocker.patch(
```
Comment (rnathuji) [inline backend/tests/test_endpoints.py:74]: Minor: You can make a small change to this test to also assert the expected change is made to the session store:

```python
    session_store = {"12345": "ewfnweoif"}
    mocker.patch(
        "rope.api.sessions.sessions",
        session_store,
    )
    response = test_client.delete("/session")
    assert response.status_code == 200
    assert session_store.get("12345") is None
```

File: backend/api/sessions.py [hunk 2]
```diff
@@ -0,0 +1,19 @@
+from fastapi import Request
+
+sessions = {}
```
Comment (rnathuji) [inline backend/api/sessions.py:3]: Would it help avoid future confusion if we named this global something like `session_store` or similar?

**Review by rnathuji — APPROVED** [review]

**Pull request merged**